File: backend/src/main/java/com/example/backend/enums/NovelStatus.java
package com.example.backend.enums;

import com.example.backend.exception.InvalidNovelStatusException;
import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public enum NovelStatus {
    DANG_TIEN_HANH("Đang tiến hành"),
    TAM_NGUNG("Tạm ngưng"),
    DA_HOAN_THANH("Đã hoàn thành");

    private final String status;

    public static NovelStatus fromValue(String value) {
        for (NovelStatus status : values()) {
            if (status.status.equals(value)) {
                return status;
            }
        }
        throw new InvalidNovelStatusException(value);
    }
}

File: frontend/src/models/NovelStatus.ts
export enum NovelStatus {
  Ongoing = "Đang tiến hành",
  Paused = "Tạm ngưng",
  Completed = "Đã hoàn thành"
}

export const mapNovelStatusToApiValue = (novelStatus: NovelStatus): string => {
  switch (novelStatus) {
    case NovelStatus.Ongoing:
      return "DANG_TIEN_HANH";
    case NovelStatus.Paused:
      return "TAM_NGUNG";
    case NovelStatus.Completed:
      return "DA_HOAN_THANH";
    default:
      return "";
  }
};

export const mapApiStatusToNovelStatus = (apiStatus: string): NovelStatus => {
  switch (apiStatus) {
    case "DANG_TIEN_HANH":
      return NovelStatus.Ongoing;
    case "TAM_NGUNG":
      return NovelStatus.Paused;
    case "DA_HOAN_THANH":
      return NovelStatus.Completed;
    default:
      return apiStatus as NovelStatus;
  }
};